client/src/TreeMap/TreeMapControls.tsx

```diff
@@ -148,7 +180,7 @@ export class TreeMapControls extends React.Component {
       key,
       value
     );
-    if (location.pathname !== new_path) {
+    if (location && history && location.pathname !== new_path) {
       history.push(new_path);
     }
```

Comment:
I think this would be a good place to use the `!` non-null operator. We're guaranteed that `location` and `history` will exist at this point, but checking them in the conditional might imply to future devs that they might not (when really that's just there to make TS happy).

There have been some cases where I've recommended type narrowing with a conditional to avoid having to spam `!`s all over the place. I _think_ I'd still advise that in some cases, but only if it's a longer code block and the `!`s are really fiddly/annoying.